addons/mail/static/src/components/message/message.js

```diff
@@ -302,6 +304,14 @@ class Message extends Component {
                 if (value.new_value !== undefined){
                     value.new_value = value.new_value ? this.env._t("True") : this.env._t("False");
                 }
+            } else if (value.field_type === 'monetary' && value.currency_id) {
+                const currency_id = session.currencies[value.currency_id]
+                if (value.old_value !== false) {
```

Comment:
I understand that you don't want to just `if (value.old_value)` because that would exclude 0, and I'm not sure where you get `false` from, but "unset" in JS should be `undefined`, not `false`.